### PR TITLE
Type hints added

### PR DESCRIPTION
**Pull request by DanielR59**
Typing hints was added to automl.py for linting with VSCode

2022-02-28_08h02_10

### CODE REVIEW

File: supervised/automl.py [hunk 1]
```diff
@@ -318,7 +343,13 @@ def __init__(
         self.n_jobs = n_jobs
         self.random_state = random_state
 
-    def fit(self, X, y, sample_weight=None, cv=None):
+    def fit(
+        self,
+        X: Union[numpy.ndarray, pandas.DataFrame],
+        y: Union[numpy.ndarray, pandas.DataFrame],
```
Comment (pplonski) [inline supervised/automl.py:349]: `y` should be `Union[numpy.ndarray, pandas.Series]`

File: supervised/model_framework.py [hunk 2]
```diff
@@ -503,6 +503,8 @@ def save(self, results_path, model_subpath):
                 "train_time": self.get_train_time(),
                 "is_stacked": self._is_stacked,
             }
+            if type(desc["final_loss"]) == np.float32:
```
Comment (pplonski) [inline supervised/model_framework.py:506]: why this change?

**Review by pplonski — CHANGES_REQUESTED** [review]
thank you for PR! looks fantastic! 

Please see minor comments in the code.

File: supervised/model_framework.py (hunk 2 above)
Comment (DanielR59) [reply supervised/model_framework.py:506]: For solving #496 issue

**Comment by pplonski** [comment]
Thank you @DanielR59!!! Great job!
